pageserver/src/tenant/mgr.rs

```diff
@@ -1970,12 +1979,28 @@ impl TenantManager {
             shard_identity,
             None,
             SpawnMode::Eager,
+            Some(&attempt),
             ctx,
         );
 
-        slot_guard.upsert(TenantSlot::Attached(tenant))?;
+        slot_guard.upsert(TenantSlot::Attached(tenant.clone()))?;
+
+        if reparented_all {
```

Comment:
`if reparented.is_empty()` AND there was no detach the restart could be avoided.